go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -68,6 +69,18 @@ func (bsnc *BaseSecondaryNetworkController) AddSecondaryNetworkResourceCommon(ob
 		}
 		return bsnc.AddNamespaceForSecondaryNetwork(ns)
 
+	case factory.PolicyType:
+		np, ok := obj.(*knet.NetworkPolicy)
+		if !ok {
+			return fmt.Errorf("could not cast %T object to *knet.NetworkPolicy", obj)
+		}
+
+		if err := bsnc.addNetworkPolicy(np); err != nil {
+			klog.Infof("Network Policy add failed for %s/%s, will try again later: %v",
+				np.Namespace, np.Name, err)
+			return err
+		}
+
```

Comment:
I guess these could live in the base event handler if they are going to be handled equally for all network types.

Reply:
all the secondary networks event handling done here, similarly for default network, it's being handled in default network controller.  do you mean to say PolicyType handling can be moved into BaseNetworkController as policy event handling is common for both ?